UI/src/view/TrafficSignReadOnlyLayer.js

```diff
@@ -0,0 +1,151 @@
+(function (root) {
+  root.TrafficSignReadOnlyLayer = function(params) {
+    var allowGrouping = true,
+      parentLayerName = params.layerName,
+      style = params.style,
+      application= applicationModel,
+      assetLabel = params.assetLabel,
+      collection = params.collection,
+      assetGrouping = params.assetGrouping,
+      map = params.map;
+
+    var me = this;
+    me.minZoomForContent = zoomlevels.minZoomForAssets;
+    var vectorSource = new ol.source.Vector();
+    var vectorLayer = new ol.layer.Vector({
+      source: vectorSource,
+      style: function (feature) {
+        return style.browsingStyleProvider.getStyle(feature);
+      },
+      renderBuffer: 0
+    });
+
+    var isShowingForLayer = {
+      speedLimit: false,
+      trSpeedLimits: false
+    };
+
+    var setLayerToShow = function(layerName, isShowing){
+      isShowingForLayer[layerName] = isShowing;
+    };
+
+    vectorLayer.set('name', 'trafficSignReadOnly' + parentLayerName);
+    vectorLayer.setOpacity(1);
+    vectorLayer.setVisible(false);
+    map.addLayer(vectorLayer);
+
+    var showSpeedLimitSigns = function(){
+      setLayerToShow(parentLayerName, true);
+      collection.setTrafficSigns('speedLimits', true);
+      me.show();
+      me.refreshView();
+    };
+
+    var hideSpeedLimitSigns = function(){
+      setLayerToShow(parentLayerName, false);
+      collection.setTrafficSigns('speedLimits', false);
+      me.hide();
+    };
+
+    this.unHighLightLayer = function(){
+      vectorLayer.setOpacity(0.15);
+    };
+
+    this.highLightLayer = function(){
+      vectorLayer.setOpacity(1);
+    };
+
+    eventbus.on(parentLayerName + ':hideSpeedLimitsSigns', hideSpeedLimitSigns);
+    eventbus.on(parentLayerName + ':showSpeedLimitsSigns', showSpeedLimitSigns);
+    eventbus.on('readOnlyLayer:' + parentLayerName + ':shown', function (layerName) {
+       showLayer(layerName);
+    }, this);
+
+    var showLayer = function(layer){
+      if(layer == parentLayerName && isShowingForLayer[parentLayerName]){
+        me.show();
+        me.refreshView();
+      }
+    };
+
+    this.refreshView = function () {
+      collection.fetch(map.getView().calculateExtent(map.getSize())).then(function (assets) {
+          var features = (!allowGrouping) ? _.map(assets, createFeature) : getGroupedFeatures(assets);
+          vectorLayer.getSource().addFeatures(features);
+          vectorLayer.getSource().addFeatures(assetLabel.renderFeaturesByPointAssets(assets, map.getView().getZoom()));
+          applySelection();
+        });
+    };
+
+    this.removeLayerFeatures = function() {
+      vectorLayer.getSource().clear();
+    };
+
+    var selectControl = new SelectToolControl(application, vectorLayer, map, {
```

Comment:
Probably this shouldn't exist since there isn't any interaction associated to this layer